packages/core/content-releases/admin/src/constants.ts

```diff
@@ -1,57 +1,74 @@
 import { Permission as StrapiPermission } from '@strapi/helper-plugin';
 
-type Permission = Pick<StrapiPermission, 'action' | 'subject'>;
-interface PermissionMap {
-  main: Permission[];
-  create: Permission[];
-  update: Permission[];
-  delete: Permission[];
-  createAction: Permission[];
-  deleteAction: Permission[];
-  publish: Permission[];
-}
-
-export const PERMISSIONS: PermissionMap = {
+export const PERMISSIONS: Record<string, StrapiPermission[]> = {
```

Comment:
I wonder why we just had action and subject before?

Also the previous types were a bit more strict, since we know every possible key I think there was some value in having them

Reply:
Regarding the keys, I think this is actually a great use case for the satisfies keyword: https://www.totaltypescript.com/clarifying-the-satisfies-operator

Reply:
before we added just action and subject because are the only properties we need to fill, but now, using the useRBAc to retrieve the permissions I need to pass an object with a key and a value in the shape of StrapiPermission because it is what useRBAC wants

Reply:
Makes sense. But I think my point about the satisfies keyword still stands, because with your changes we lose type safety for key names

Reply:
thank you @remidej for the advice